build/ci/github_actions/driver/deploy_operator.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -xel
+set +o pipefail
+
+edit_operator_yaml_image (){
+  cd $(dirname $operator_yaml)
+  operator_image_in_branch=`yq eval '(. | select(.kind == "Deployment") | .spec.template.spec.containers[0].image)' $(basename $operator_yaml)`
+  sed -i "s+$operator_image_in_branch+$operator_image_for_test+g" $(basename $operator_yaml) ## TODO: CSI-3223 avoid using sed
+cd -
+}
+
+edit_operator_yaml_image
+cat $operator_yaml | grep image:
+kubectl apply -f $operator_yaml
+. wait_for_pods_ready_utils.sh && wait_for_pod_to_start "operator"
+. wait_for_pods_ready_utils.sh && assert_expected_image_in_pod "operator" $operator_image_for_test
+. wait_for_pods_ready_utils.sh && wait_for_driver_deployment_to_finish
```

Comment:
what does the driver deployment have to do with `deploy_operator.sh`?

Reply:
technically, I'm not sure the operator is really "csi".
can't it just wait specifically for the operator pod?

Reply:
why creating two functions that do the approximately the same

Reply:
because we don't really need 1 function to check "all" pods with "csi" label, we need 2 functions: 1 that only checks the operator pod, and 1 that only checks the driver pods.

it's not so bad that the operator pod is also checked when we want to check the driver pods (because the operator should be running anyway), but (potentially) checking the driver pods when we want to check the operator pod is not good.

Reply:
btw we have a special label for the operator pod (see [1.d.](https://www.ibm.com/docs/en/stg-block-csi-driver/1.6.0?topic=driver-installing-github))

Reply:
done